tokenizers/src/models/bpe/trainer.rs

```diff
@@ -502,6 +502,22 @@ impl BpeTrainer {
                 }
             }
             let new_token = format!("{}{}", part_a, part_b);
+            // implement sentencepiece-like merge.
+            // if this code were to be merged, integrate a way in the python bindings to communicate this variable
+            // default should be 0/None to maintain previous behavior. 16 is the spm default.
+            let max_merge_length: Option<u16> = Some(16);
+
+            match max_merge_length {
+                None | Some(0) => {
+                // in case 0 was manually entered, treat as None
+                }
+                Some(length) => {
+                    if new_token.chars().count() >
+                    (length as usize) {
+                        continue;
+                    }
+                }
+            }
```

Comment:
```suggestion
           if let Some(max_token_length) = max_token_length{
               if new_token.chars().count() > max_token_length{
                   continue;
               }
           }
```

This is more idiomatic imo.
0 is NOT a special value. `None`  means ignore, `zero` does mean zero. If things starts to do weird things it's not the problem of this function, it's respecting the value which is more important.

Try to switch to `usize` it makes everything easier to read, and the actual "size" isn't important optimization wise.